### ISSUE TITLE
Error after updating to 3.9.0

### ISSUE THREAD

**Issue by aa12stepoffice** [code4recovery/12-step-meeting-list#203]
I installed the latest plugin update and it caused an error which crashed my site (running WordPress 5.5.3 and viewing site on google chrome):

An error of type E_ERROR was caused in line 33 of the file /home/aamelbou/public_html/wp-content/plugins/12-step-meeting-list/includes/admin_meeting.php. Error message: Uncaught Error: Call to undefined function db_update_addresses_cache_approximate_location() in /home/aamelbou/public_html/wp-content/plugins/12-step-meeting-list/includes/admin_meeting.php:33 Stack trace: #0 /home/aamelbou/public_html/wp-includes/class-wp-hook.php(287): tsml_admin_init('') #1 /home/aamelbou/public_html/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters(NULL, Array) #2 /home/aamelbou/public_html/wp-includes/plugin.php(478): WP_Hook->do_action(Array) #3 /home/aamelbou/public_html/wp-admin/admin.php(175): do_action('admin_init') #4 /home/aamelbou/public_html/wp-admin/index.php(10): require_once('/home/aamelbou/...') #5 {main} thrown

I rolled back to version 3.8.0 which restored the site but the plugin is still not working. When I visit https://aamelbourne.org.au/meetings it redirects to https://aamelbourne.org.au/tsml_group/meetings-with-bill-by-the-bay-group/. 

Please let me know if I have not provided enough information.

Michael K

**Comment by julien-paris** [comment]
Hi folks, exact same problem here. 
I was also locked out of admin.
I disabled the plugin by renaming it in FTP and the site is back up, but without the meetings.

**Comment by richieboo** [comment]
Same at South Jersey Intergroup. With WordPress 5.5.3, entering recovery mode, I can see the same error reported by aa12stepoffice when I view Plugins on Admin side, is the error that is reported.

**Comment by Bdrv** [comment]
Same at Miami Valley Ohio Intergroup (Dayton). Running WP 5.5.3, had to disable 12-Step Meeting plugin to restore website. https://al-anondaytonoh.org

**Comment by k8fish3r** [comment]
Is a fix coming ? Should we each restore earlier point and disable these automatic updates coming through to stop 12sml sending the update again?

Thankful for the programme existing but hoping to save out website being taken down by the plugin is top priority,

**Comment by k8fish3r** [comment]
We're at "critical error on your website territory - login to admin but not much more

**Comment by tech2serve** [comment]
I'm looking at this now. Based on the report, I believe I know *what* is happening, just not sure why since this did not occur in testing.

**Comment by eviaatech** [comment]
We are experiencing the same error. Plugin is currently disabled on our site as well.

**Comment by eviaatech** [comment]
eviaaweb.org is our site

**Comment by Grateful-Viv** [comment]
Same here. Started this on a staging site by rolling back to a version from 2 weeks ago. Everything installed fine and was able to see the front end, but could no longer get to any admin back end areas.

**Comment by tech2serve** [comment]
@Grateful-Viv, Glad you're here. In a few minutes, I'm going to release `v3.9.1-b`, and would appreciate if you could test it?

**Comment by richieboo** [comment]
I can test it too.. for aasj.org - tech@aasj.org

**Comment by tech2serve** [comment]
For those who already know how to manually install plugins, you can try [this beta version](https://github.com/code4recovery/12-step-meeting-list/releases/tag/v3.9.1-b). Please report results as soon as you can.

For those who aren't familiar with that process, please be patient as we try to fix the issue. Regrettably, as the issue didn't exist on my site (or that of our initial testers), I'm not 100% confident of the cause.

However, this will remain my priority today until the issue is resolved.

**Comment by petermixt** [comment]
For now... I activated recovery mode. I then disabled the plug-in.  Then download the 3.9.0 version here in GitHub and install it. The new 3.9 installed/loaded properly for me. Then I deleted the bad version.  Done. (for now)

**Comment by tech2serve** [comment]
@petermixt Please confirm which version you downloaded for GitHub that worked?

**Comment by petermixt** [comment]
version that worked for me = 3.9.0 on [github ](https://github.com/code4recovery/12-step-meeting-list)
I see you now have a 3.9.1b version too

**Comment by richieboo** [comment]
12-step-meeting-list-3.9.1-b worked for https://aasj.org/meetings !!! 

Thank you!

**Comment by tech2serve** [comment]
@petermixt, 

It sounds like you pulled the zip using this link?

<img width="936" alt="image" src="https://user-images.githubusercontent.com/31904316/97783138-bbb75a80-1b63-11eb-99ab-2e35439e4f98.png">

If so, you'd have actually gotten v3.9.1-b, and that + @richieboo's comment would seem to have resolved the issue. 

I'll get this bundled up as a release to Wordpress.

**Comment by petermixt** [comment]
pulled it yesterday... it read as 3.9.0.  But yes the 3.9.1-b is a charmer. Again thank you for a WONDERFUL plug-in.

**Comment by eviaatech** [comment]
3.9.1-b fixed meeting page, but theme/appearance is affected. Debugging some more. Home page went to white screen with text. Footer is black now.

**Comment by richieboo** [comment]
> 3.9.1-b fixed meeting page, but theme/appearance is affected. Debugging some more. Home page went to white screen with text. Footer is black now.

Your console log seems to be complaining about the Divi customizer. Might try to customize something again to rewrite some caching?

**Comment by eviaatech** [comment]
> > 3.9.1-b fixed meeting page, but theme/appearance is affected. Debugging some more. Home page went to white screen with text. Footer is black now.
> 
> Your console log seems to be complaining about the Divi customizer. Might try to customize something again to rewrite some caching?

I customized a couple things, but I do not know how to rewrite any caching. Still getting the error. Even after disabling and reinstalling the plugin, the theme errors are present.

**Comment by tech2serve** [comment]
I have released v3.9.1 to Wordpress. If you're able, please pull that down and make sure the plugin loads correctly (i.e., no `error of type E_ERROR was caused in line 33`).

I do apologize for the inconvenience. Our team will discuss today what wasn't sufficient in our process to see if we can determine the root cause. As I said, this was unexpected as none of us, nor our gracious beta testers, experienced the issue with the last release candidate.

The error noted in this issue appears to have been related to how some functions were being loaded, and because of this path error, the plugin didn't load correctly. It is *very* (*extremely*) unlikely that this issue _directly_ caused any other problems with your site. If you continue to experience problems not related to the initial bug report, please look into other causes. I'm happy to help you and we won't rule out other issues with the plugin.

Thank you all for participating with me this morning to get this resolved. We're always looking for more beta testers. Please see our [Beta Testing Guide](https://github.com/code4recovery/12-step-meeting-list/wiki/Beta-Testing-Guide).

**Comment by tech2serve** [comment]
> 3.9.1-b fixed meeting page, but theme/appearance is affected. Debugging some more. Home page went to white screen with text. Footer is black now.

@eviaatech, I'm inclined to agree with @richieboo:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/31904316/97783683-67ae7500-1b67-11eb-89ef-3b495b266087.png">

In looking at your site, you seem to have an issue with cache or the theme itself. If I've found the right theme online, [this link](https://divitheme.net/clear-divi-cache/) might help explain how to clear the cache?

**Comment by tech2serve** [comment]
> 3.9.1-b fixed meeting page, but theme/appearance is affected. Debugging some more. Home page went to white screen with text. Footer is black now.

@eviaatech, I'm inclined to agree with @richieboo:

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/31904316/97783683-67ae7500-1b67-11eb-89ef-3b495b266087.png">

In looking at your site, you seem to have an issue with cache or the theme itself. 

> I rolled back to version 3.8.0 which restored the site but the plugin is still not working. When I visit https://aamelbourne.org.au/meetings it redirects to https://aamelbourne.org.au/tsml_group/meetings-with-bill-by-the-bay-group/.

@aa12stepoffice, please go to Wordpress Settings-->Permalinks, and scroll down and then "save". That should refresh the links, and hopefully cause your meetings to display again.

**Comment by eviaatech** [comment]
> > 3.9.1-b fixed meeting page, but theme/appearance is affected. Debugging some more. Home page went to white screen with text. Footer is black now.
> 
> @eviaatech, I'm inclined to agree with @richieboo:
> 
> <img alt="image" width="1440" src="https://user-images.githubusercontent.com/31904316/97783683-67ae7500-1b67-11eb-89ef-3b495b266087.png">
> 
> In looking at your site, you seem to have an issue with cache or the theme itself. If I've found the right theme online, [this link](https://divitheme.net/clear-divi-cache/) might help explain how to clear the cache?

Cleared the cache. Turned off the CSS Static File Generation. Opened site. Turned it back on. All is good. Thank you for web servants with more experience than I! Thanks to tech2serve for quick response to the issue and a fix!

**Comment by tech2serve** [comment]
Again, I apologize for the the troubles. Luckily, because of our friends in other time zones, the issue was caught quickly and overall the impact will (pray-fully) be minimal.

You will see a `v3.9.2` get released here shortly. This is really just a version bump because I missed updating one of the version numbers in the readme.txt, so the incorrect version is showing on Wordpress. Sigh....

**Comment by eviaatech** [comment]
Thanks for all you do to help us carry the message!

**Comment by machouinard** [comment]
Just ran into this issue, trying to catch up.  

Still having trouble with 3.9.2.  Seems there is no `includes/database_updates.php` file in the WP repo, either in trunk or /tags/3.9.2

https://plugins.svn.wordpress.org/12-step-meeting-list/tags/3.9.2/includes/

**Comment by OIG-Webmaster** [comment]
The latest version 3.9.2 just killed our site.  I've had to go in through a term and remove it. I was hoping the version you have up now would fix it, but apparently, it's the problem.

**Comment by virtualtom** [comment]
Yes, it appears that the install of 3.9.2 is missing /includes/database_updates.php.  I get these errors on activation:

Warning: include_once(/homepages/18/d767263127/htdocs/app793772400/wp-content/plugins/12-step-meeting-list//includes/database_updates.php): failed to open stream: No such file or directory in /homepages/18/d767263127/htdocs/app793772400/wp-content/plugins/12-step-meeting-list/includes/functions.php on line 4

Warning: include_once(): Failed opening '/homepages/18/d767263127/htdocs/app793772400/wp-content/plugins/12-step-meeting-list//includes/database_updates.php' for inclusion (include_path='.:/usr/lib/php7.3') in /homepages/18/d767263127/htdocs/app793772400/wp-content/plugins/12-step-meeting-list/includes/functions.php on line 4

Warning: Cannot modify header information - headers already sent by (output started at /homepages/18/d767263127/htdocs/app793772400/wp-content/plugins/12-step-meeting-list/includes/functions.php:4) in /homepages/18/d767263127/htdocs/app793772400/wp-includes/functions.php on line 6271

Fatal error: Uncaught Error: Call to undefined function db_update_addresses_cache_approximate_location() in /homepages/18/d767263127/htdocs/app793772400/wp-content/plugins/12-step-meeting-list/includes/admin_meeting.php:33 Stack trace: #0 /homepages/18/d767263127/htdocs/app793772400/wp-includes/class-wp-hook.php(287): tsml_admin_init('') #1 /homepages/18/d767263127/htdocs/app793772400/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters(NULL, Array) #2 /homepages/18/d767263127/htdocs/app793772400/wp-includes/plugin.php(478): WP_Hook->do_action(Array) #3 /homepages/18/d767263127/htdocs/app793772400/wp-admin/admin.php(175): do_action('admin_init') #4 /homepages/18/d767263127/htdocs/app793772400/wp-admin/plugins.php(10): require_once('/homepages/18/d...') #5 {main} thrown in /homepages/18/d767263127/htdocs/app793772400/wp-content/plugins/12-step-meeting-list/includes/admin_meeting.php on line 33
There has been a critical error on your website. Please check your site admin email inbox for instructions.

**Comment by Trundlore** [comment]
getting the same error with 3.9.2. not sure if we had the problem with 3.9.0

An error of type E_ERROR was caused in line 33 of the file /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/admin_meeting.php. Error message: Uncaught Error: Call to undefined function db_update_addresses_cache_approximate_location() in /srv/htdocs/wp-content/plugins/12-step-meeting-list/includes/admin_meeting.php:33 Stack trace: #0 /wordpress/core/5.5.3/wp-includes/class-wp-hook.php(287): tsml_admin_init('') #1 /wordpress/core/5.5.3/wp-includes/class-wp-hook.php(311): WP_Hook->apply_filters(NULL, Array) #2 /wordpress/core/5.5.3/wp-includes/plugin.php(478): WP_Hook->do_action(Array) #3 /wordpress/core/5.5.3/wp-admin/admin.php(175): do_action('admin_init') #4 /wordpress/core/5.5.3/wp-admin/plugins.php(10): require_once('/wordpress/core...') #5 {main} thrown

**Comment by tech2serve** [comment]
@Trundlore @virtualtom --

Please try v3.9.3.

**Comment by Trundlore** [comment]
> @Trundlore @virtualtom --
> 
> Please try v3.9.3.

seems to have sorted it.
many thanks for your service

**Comment by virtualtom** [comment]
Version 3.9.3 fixed my problems!

Thank you for the fast work!

**Comment by eviaatech** [comment]
All is functioning as expected. Phew!!

**Comment by julien-paris** [comment]
3.9.3 is a full fix for me. 
That was resolved at lightening speed. Thanks so much!

**Comment by district6wc** [comment]
I'm still getting these errors and I am unable to log into the wp dashboard. I have deactivated the plugin, but version is not being recognized back to 3.9.0  Please help...

Warning: include_once(/home/izgr1zclaxch/public_html/wp-content/plugins/12-step-meeting-list/includes/database_updates.php): failed to open stream: No such file or directory in /home/izgr1zclaxch/public_html/wp-content/plugins/12-step-meeting-list/includes/functions.php on line 3

Wa

**Comment by aa12stepoffice** [comment]
Just installed version 3.9.3 and our meetings page is fixed!

@tech2serve Thank you for the speedy response, and for your service.

**Comment by kcristiano** [comment]
I've had the same issues as desribed by @district6wc 

I updated and had the fatal error, rolled back.   I then tried 3.9.2 and 3.9.3 via the WP built in update system.

On both versions it was missing the file `12-step-meeting-list//includes/database_updates.php` 

I downloaded the file directly from `https://downloads.wordpress.org/plugin/12-step-meeting-list.3.9.3.zip` and compared to the master branch of github.   The file was there.

I then updated to using a git checkout and version 3.9.3 works fine.

WP 5.4.4, php 7.4 is what we are using.

I have no idea why the WP updater is causing the file to not be included, but I do think getting a copy from github or downloading directly from the plugin repo is the ay to go if you are having issues.

**Comment by tech2serve** [comment]
> I'm still getting these errors and I am unable to log into the wp dashboard. I have deactivated the plugin, but version is not being recognized back to 3.9.0 Please help...
> 
> Warning: include_once(/home/izgr1zclaxch/public_html/wp-content/plugins/12-step-meeting-list/includes/database_updates.php): failed to open stream: No such file or directory in /home/izgr1zclaxch/public_html/wp-content/plugins/12-step-meeting-list/includes/functions.php on line 3
> 
> Wa

@district6wc, I think some screen caps of what you're seeing would help me. In particular, I'm confused by the statements that you're not able to log into the Wordpress dashboard, but that you disabled the plugin. 

If you're able to log into the admin side of your Wordpress site, and you should be able to, then try performing these steps:

1. Click on Plugins
1. Determine which version of the TSML plugin is installed

    ![image](https://user-images.githubusercontent.com/31904316/97789219-a4409780-1b8c-11eb-97bb-794b5b34373b.png)

1. If not showing 3.9.3, please update to latest version.

Let me know where you stand at that point, please.

**Comment by tech2serve** [comment]
> I have no idea why the WP updater is causing the file to not be included, but I do think getting a copy from github or downloading directly from the plugin repo is the ay to go if you are having issues.

@kcristiano Thanks for the comment. I do _now_ know why the missing file was not being included through the Wordpress site; the short version is that it was my deployment process error.

However, the v3.9.3 on the Wordpress site is correct and includes all the necessary files. Users ought to be able to update their sites to v3.9.3 now using Wordpress. Nobody should need to manually drop any files in place from GitHub. 

Well, that is how is _should_ be now :-)

**Comment by district6wc** [comment]
![12steppluginversion](https://user-images.githubusercontent.com/1336342/97789358-326d4b80-1b96-11eb-8ba5-014faff02b5a.PNG)


I have pasted a screenshot of plugin.  when I activate, I can no longer access the wp-admin.  Right now, I have renamed the 12 Step Meeting List folder so I can get to the dashboard.  I apologize if I am not explaining it correctly.

my website is new.alanonalateen6nc.org

**Comment by tech2serve** [comment]
@district6wc That is correct. Please update to v3.9.3 using "update now" first. Then activate the plugin, and all should be good.

**Comment by district6wc** [comment]
> @district6wc That is correct. Please update to v3.9.3 using "update now" first. Then activate the plugin, and all should be good.

plugin is activated and I see this error on page and can no longer get to plugin from the dashboard.:

![image](https://user-images.githubusercontent.com/1336342/97790037-c7bf0e80-1b9b-11eb-91a8-52e59c47f27f.png)

Here is the some of the contents of the /includes/function.php file line 3

<?php

include_once dirname(__FILE__) . '/database_updates.php';

//function: helper for debugging
//used:		ad-hoc
if (!function_exists('dd')) {
	function dd($array) {
		echo '<pre>';
		print_r($array);
		exit;
	}
}

**Comment by tech2serve** [comment]
Hmmm. Very strange that `functions.php` is missing as that isn't the file we had problems with earlier. 

Who is your hosting provider? Are you able to access files through ftp or your hosting platform c-panel or dashboard?

**Comment by district6wc** [comment]
yes, I have been able to access through cpanel/file manager.  the functions.php file is not missing.

**Comment by tech2serve** [comment]
My mistake.

In reading the message, I'm suspecting your `functions.php` file is not getting updated with the Wordpress update process. Are you able to inspect the `functions.php` file?

Admittedly, _and assuming you have a good backup of your meeting information_, the easiest and quickest solution for you might be to delete the `12-step-meeting-list` folder from your plugins directory, and reinstall it. Actually, that should affect your meeting data, but it is always best to have a back up of the data handy.

**Comment by machouinard** [comment]
I'd make sure the plugin is actually updated to 3.9.3.  The error is that in line 3 of `functions.php` it's trying to load `database_updates.php` and that's the file it's not finding ( because it wasn't there in earlier versions ).

**Comment by tech2serve** [comment]
Yeah. According to the error shown in the screen cap, the problem include is on line 3. However, in `v3.9.3`, that include moved to line 4.

So, it does appear something on @district6wc's site is not actually updating the plugin even though it may be reporting as such.

**Comment by machouinard** [comment]
Can always change the plugin version in the main file to 3.9.1 to force the update availability.

Is there any hardcore caching on the site?

**Comment by district6wc** [comment]
I have downloaded the new version. replaced function.php file with new one. now error is showing on line 4.

 https://downloads.wordpress.org/plugin/12-step-meeting-list.3.9.3.zip.

**Comment by machouinard** [comment]
@district6wc did you replace the entire plugin folder or just the `functions.php` file?

**Comment by tech2serve** [comment]
If you *only* replaced the `functions.php`, that won't work. 

Your best bet at this stage is to delete the plugin folder and then install from scratch. Your data **should** be untouched.

**Comment by district6wc** [comment]
ok, I will do that after I take a break.  Thank you so much!!  I love the plug-in!!

**Comment by tech2serve** [comment]
I think we've addressed all the concerns raised in this thread, so I'm going to close it.

If I missed anyone, please open a new issue.

**Comment by district6wc** [comment]
Just wanted to let you know that I reinstalled the plugin and my data was
gone.  Luckily, I had exported a current meeting list back on October 22
that I imported back in.  my data was back with only a few changes to add
back in.  I wished I had done an export before reinstalling the plugin.
Also Wordpress updated as well, so not sure which one caused the issue.

Thanks so  much for your help!

Yours in Service,
Stephanie Holopainen
Learning To Live AFG, Graham

District 6 Website Coordinator Panel 59
336 675-1301
